### DISCUSSION TITLE
What scripts are run for TPCC test (PostgreSQL)?

### DISCUSSION THREAD

**Discussion by miashaee** [TPC-Council/HammerDB#382] [Q&A] [answered]
Hi, I was able to set up some Hammerdb testing against my PostgreSQL database, but I am wondering how you go about seeing the exact queries that are being executed against the DB and how often, I am just using the default test scripts from HammerDB. This would help me to understand what is running.

**Comment by sm-shaw** [comment, marked as answer]
What you need to understand the workload that is running is the [PostgreSQL metrics viewer](https://www.hammerdb.com/docs/ch07s05.html)

This will show you the exact SQL that is running, as well as what takes the most time and statistics.

![pgstats](https://user-images.githubusercontent.com/38044085/178308412-d59e34ce-5207-4864-b082-d9a8b2320b61.png)

**Reply by miashaee** [reply]
Thanks, but if I'm running command-line how would I see all of this? Is there a setting that I'd have to use for that? Thanks.

**Reply by sm-shaw** [reply]
Yes, if you are proficient in SQL use the pg_stat_statements and pg_sentinel packages (This is what HammerDB does) to run SQL queries to see the same information.  If not particularly proficient in SQL then build an installation on a local GUI enabled PC either Windows or Linux and run the GUI - the workload that HammerDB is running is exactly the same.
